qiskit_nature/second_q/circuit/library/initial_states/hartree_fock.py

```diff
@@ -15,62 +15,189 @@
 
         Raises:
             TypeError: If qubit_converter contains BravyiKitaevSuperFastMapper. See
                 https://github.com/Qiskit/qiskit-nature/issues/537 for more information.
         """
-        if isinstance(qubit_converter.mapper, BravyiKitaevSuperFastMapper):
-            raise TypeError(
-                "Unsupported mapper in qubit_converter: ",
-                type(qubit_converter.mapper),
-                ". See https://github.com/Qiskit/qiskit-nature/issues/537",
+
+        super().__init__()
+        self._qubit_converter = qubit_converter
+        self._num_spatial_orbitals = num_spatial_orbitals
+        self._num_particles = num_particles
+        self._bitstr: list[bool] | None = None
+
+        if qubit_converter is not None:
+            self.qubit_converter = qubit_converter
+
+        if num_spatial_orbitals is not None:
+            self.num_spatial_orbitals = num_spatial_orbitals
+
+        if num_particles is not None:
+            self.num_particles = num_particles
+
+    @property
+    def qubit_converter(self) -> QubitConverter:
+        """The qubit operator converter."""
+        return self._qubit_converter
+
+    @qubit_converter.setter
+    def qubit_converter(self, conv: QubitConverter) -> None:
+        """Sets the qubit operator converter."""
+        self._invalidate()
+        self._qubit_converter = conv
+        self._reset_register()
+
+    @property
+    def num_spatial_orbitals(self) -> int:
+        """The number of spatial orbitals."""
+        return self._num_spatial_orbitals
+
+    @num_spatial_orbitals.setter
+    def num_spatial_orbitals(self, n: int) -> None:
+        """Sets the number of spatial orbitals."""
+        self._invalidate()
+        self._num_spatial_orbitals = n
+        self._reset_register()
+
+    @property
+    def num_particles(self) -> tuple[int, int]:
+        """The number of particles."""
+        return self._num_particles
+
+    @num_particles.setter
+    def num_particles(self, n: tuple[int, int]) -> None:
+        """Sets the number of particles."""
+        self._invalidate()
+        self._num_particles = n
+        self._reset_register()
+
+    # pylint: disable=too-many-return-statements
+    def _check_configuration(self, raise_on_failure: bool = True) -> bool:
+        """Check if the configuration of the HartreeFock class is valid.
+        Args:
+            raise_on_failure: Whether to raise on failure.
+        Returns:
+            True, if the configuration is valid and the circuit can be constructed. Otherwise
+            an ValueError or TypeError is raised.
+        Raises:
+            ValueError: If the number of spatial orbitals is not specified or equal or less than zero.
+            ValueError: If the number of particles is not specified or less than zero.
+            ValueError: If the number of spatial orbitals specified is less than or equal to the
+                        total number of particles.
```

Comment:
I have updated the check to verify that each particle number is less than the number of spatial orbitals, and its corresponding docstring.  Since invalid configurations like `num_particles = (4,2)` and `num_spatial_orbitals=3` would bypass the previous check.